packages/core/src/node-execution-context/local-load-options-context.ts

```diff
@@ -0,0 +1,64 @@
+import { get } from 'lodash';
+import { ApplicationError, deepCopy, Workflow } from 'n8n-workflow';
+import type {
+	INodeParameterResourceLocator,
+	IWorkflowExecuteAdditionalData,
+	NodeParameterValueType,
+	ILocalLoadOptionsFunctions,
+	IWorkflowLoader,
+	IWorkflowNodeContext,
+	INode,
+	INodeTypes,
+} from 'n8n-workflow';
+
+import { LoadWorkflowNodeContext } from './workflow-node-context';
+
+export class LocalLoadOptionsContext implements ILocalLoadOptionsFunctions {
+	constructor(
+		private nodeTypes: INodeTypes,
+		private additionalData: IWorkflowExecuteAdditionalData,
+		private path: string,
+		private workflowLoader: IWorkflowLoader,
+	) {}
+
+	async getWorkflowNodeContext(nodeType: string): Promise<IWorkflowNodeContext | null> {
+		const { value } = this.getCurrentNodeParameter('workflowId') as INodeParameterResourceLocator;
+
+		const workflowId = value as string;
+		if (!workflowId) {
+			throw new ApplicationError('No workflowId parameter defined on node!');
+		}
+
+		const dbWorkflow = await this.workflowLoader.get(workflowId);
+
+		const workflowNode = dbWorkflow.nodes.find((node) => node.type === nodeType) as INode;
+
+		if (workflowNode) {
+			const workflow = new Workflow({
+				nodes: [workflowNode],
```

Comment:
Are we creating a new Workflow object with just the "Subworkflow Trigger" node in it? Can we explain why that needs to be done (either with comments or with variable names)?